src/main/java/org/spongepowered/api/data/DataRegistration.java

```diff
@@ -40,8 +41,12 @@
      * the desired {@link DataManipulator} type to be registered.
      *
      * @return The new builder instance
+     * @deprecated Use the {@link DataGenerator} instead, this is the new
+     *    method of generating {@link DataManipulator} implementations and
```

Comment:
@gabizou If it's just going to be used by SpongeCommon, etc., why not move away from the builder and put it in the implementation. I thought you wanted to get rid of all the AbstractData classes, unless you still want to support custom implemented `DataManipulator`'s?